src/app/static/src/tests/components/adr/selectDataset.test.ts

```diff
@@ -528,6 +555,76 @@ describe("select dataset", () => {
         expect(rendered.find("#loading-dataset").exists()).toBe(false);
         expect(rendered.find(Modal).props("open")).toBe(false);
     });
+    
+    it("renders reset confirmation dialog when changing selected dataset and then modal if click save", async () => {
+        let store = getStore({
+            selectedDataset: fakeDataset
+        },
+            {datasets: [{...fakeRawDatasets[0], ...fakeRawDatasets[1], resources: [shape]}]}, true
+        )
+        const rendered = mount(SelectDataset, {
+            store, stubs: ["tree-select"]
+        });
+        rendered.find("button").trigger("click");
+
+        await Vue.nextTick();
+
+        expect(rendered.find(ResetConfirmation).exists()).toBe(true);
+        expect(rendered.find(ResetConfirmation).props("open")).toBe(true);
+        expect(rendered.find(Modal).props("open")).toBe(false);
+        const buttons = rendered.findAll("button");
+        expectTranslated(buttons.at(3), "Save version and keep editing",
+            "Sauvegarder la version et continuer à modifier", store);
+        buttons.at(3).trigger("click");
```

Comment:
If this line duplicates the next one then it should be removed

Reply:
Not quite sure what you are referring to here

Reply:
Lines 573 and 574 are identical... is that intentional?